services/_shared/personas/code_reviewer/sast.py

```
@@ -227,22 +227,37 @@ def scan_semgrep(
+            # startup. The pods run as uid 10001 created with --no-create-home
+            # on a readOnlyRootFilesystem, so that mkdir crashed semgrep with
+            # exit 1 before it scanned anything - every production scan
+            # silently degraded to zero findings (found live 2026-07-13, the
+            # infra#1776 sweep). Point HOME (+ XDG cache) at the scan's own
+            # temp dir, the one path we know is writable and gets cleaned up.
+            sem_env = {
+                **os.environ,
+                "HOME": tmp,
+                "XDG_CACHE_HOME": os.path.join(tmp, ".cache"),
+            }
             proc = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)